File: September/24/simplifypath.py
# LeetCode 71: Simplify Path
# Explanation:
# 1. Split path by '/'.
# 2. Use a stack to build valid directories.
# 3. Skip '.' and empty parts, pop for '..'.
# 4. Join stack with '/'.
# Time Complexity: O(n)
# Space Complexity: O(n)

class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = []
        for part in path.split("/"):
            if part == "" or part == ".":
                continue
            elif part == "..":
                if stack:
                    stack.pop()
            else:
                stack.append(part)
        return "/" + "/".join(stack)

File: 2025/September/24/simplifypath.cs
// LeetCode 71: Simplify Path
// Explanation:
// 1. Split by "/".
// 2. Use stack for directories.
// 3. Skip "." and empty parts, pop for "..".
// 4. Build canonical path.
// Time Complexity: O(n)
// Space Complexity: O(n)

public class Solution {
    public string SimplifyPath(string path) {
        var stack = new Stack<string>();
        foreach (var part in path.Split('/')) {
            if (part == "" || part == ".") continue;
            if (part == "..") {
                if (stack.Count > 0) stack.Pop();
            } else {
                stack.Push(part);
            }
        }
        var arr = stack.ToArray();
        Array.Reverse(arr);
        return "/" + string.Join("/", arr);
    }
}

// Example usage:
// Solution sol = new Solution();
// Console.WriteLine(sol.SimplifyPath("/home/"));       // "/home"
// Console.WriteLine(sol.SimplifyPath("/../"));         // "/"
// Console.WriteLine(sol.SimplifyPath("/home//foo/"));  // "/home/foo"
